src/hotspot/share/jvmci/jvmci_globals.cpp

```diff
@@ -79,15 +80,15 @@ bool JVMCIGlobals::check_jvmci_flags_are_consistent() {
   CHECK_NOT_SET(JVMCIHostThreads,             UseJVMCICompiler)
   CHECK_NOT_SET(LibJVMCICompilerThreadHidden, UseJVMCICompiler)
 
+  if ((UseJVMCICompiler || EnableJVMCI) &&
```

Comment:
Doesn't `UseJVMCICompiler` require `EnableJVMCI`?

Reply:
No.

Reply:
This will conflict with my change https://github.com/openjdk/jdk/pull/21069/commits/b75504633bc0f4fcecc1c08552e556b26d9ffbb9.

Reply:
No problem - I'll resolve it once your PR is merged.

Reply:
Based just off the flag docs, I think its easy to think that `+EnableJVMCI`is a prerequisite to enabling the other JVMCI flags, along the line of `+UnlockExperimentalVMOptions`. (That was for sure my newbie reading.)

Perhaps its docs ([here](https://github.com/openjdk/jdk/blob/0b8c9f6d2397dcb480dc5ae109607d86f2b15619/src/hotspot/share/jvmci/jvmci_globals.hpp#L47-L48)) should be updated to say "Defaults to true if UseJVMCICompiler is true"?

Reply:
I've pushed https://github.com/openjdk/jdk/pull/21120/commits/e26e68d9a70ee27b4e71da86ecb42dca11e9a24f to try clarify this a bit.

I know it's a little confusing and apologize. When JVMCI is no longer experimental, this should become much clearer.